### PR TITLE
reward bug fix

### PR DESCRIPTION
**Pull request by vmurahari3**
Fixing bug in reward calculation. Based on the mse formulation, all the samples in the batch get the same reward, which is not the desired behavior.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -270,6 +271,7 @@ def batch_iter(dataloader):
             # Make an image prediction after each round
             predFeatures = qBot.predictImage()
             featDist = mse_criterion(predFeatures, image)
+            featDist = torch.mean(featDist, 1)
```
Comment (nirbhayjm) [inline train.py]: `featLoss` here and in line 186 expects a scalar when optimizing the feature regression loss. Shouldn't this be a mean over all dimensions instead of just `dim=1`? Otherwise the code won't work for supervised pre-training

**Comment by vmurahari3** [comment]
I made the changes you suggested. Please have a look.
